shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/AutoIntervalShardingAlgorithmTest.java

```diff
@@ -119,4 +119,45 @@ public void assertRangeDoShardingWithGreaterTenTables() {
         assertThat(actual.size(), is(11));
     }
+
+    @Test
+    public void assertRangeDoShardingInValueWithMilliseconds() {
+        AutoIntervalShardingAlgorithm shardingAlgorithm = new AutoIntervalShardingAlgorithm();
+        shardingAlgorithm.getProps().setProperty("datetime-lower", "2020-01-01 00:00:00");
+        shardingAlgorithm.getProps().setProperty("datetime-upper", "2020-01-01 00:00:30");
+        shardingAlgorithm.getProps().setProperty("sharding-seconds", "1");
+        shardingAlgorithm.init();
+        List<String> availableTargetNames = new LinkedList<>();
+        for (int i = 0; i < 32; i++) {
+            availableTargetNames.add("t_order_" + i);
+        }
+        Collection<String> actualWithoutMilliseconds = shardingAlgorithm.doSharding(availableTargetNames,
+                new RangeShardingValue<>("t_order", "create_time", Range.closed("2020-01-01 00:00:11", "2020-01-01 00:00:21")));
+        assertThat(actualWithoutMilliseconds.size(), is(11));
+        Collection<String> actualWithOneMillisecond = shardingAlgorithm.doSharding(availableTargetNames,
+                new RangeShardingValue<>("t_order", "create_time", Range.closed("2020-01-01 00:00:11.1", "2020-01-01 00:00:21.1")));
+        assertThat(actualWithOneMillisecond.size(), is(11));
+        Collection<String> actualWithTwoMilliseconds = shardingAlgorithm.doSharding(availableTargetNames,
+                new RangeShardingValue<>("t_order", "create_time", Range.closed("2020-01-01 00:00:11.12", "2020-01-01 00:00:21.12")));
+        assertThat(actualWithTwoMilliseconds.size(), is(11));
+        Collection<String> actualWithThreeMilliseconds = shardingAlgorithm.doSharding(availableTargetNames,
+                new RangeShardingValue<>("t_order", "create_time", Range.closed("2020-01-01 00:00:11.123", "2020-01-01 00:00:21.123")));
+        assertThat(actualWithThreeMilliseconds.size(), is(11));
+    }
+
+    @Test
+    public void assertRangeDoShardingInPropertyWithMilliseconds() {
+        AutoIntervalShardingAlgorithm shardingAlgorithm = new AutoIntervalShardingAlgorithm();
+        shardingAlgorithm.getProps().setProperty("datetime-lower", "2020-01-01 00:00:00.123");
+        shardingAlgorithm.getProps().setProperty("datetime-upper", "2020-01-01 00:00:30.123");
+        shardingAlgorithm.getProps().setProperty("sharding-seconds", "1");
+        shardingAlgorithm.init();
+        List<String> availableTargetNames = new LinkedList<>();
+        for (int i = 0; i < 32; i++) {
+            availableTargetNames.add("t_order_" + i);
+        }
+        Collection<String> actualWithoutMilliseconds = shardingAlgorithm.doSharding(availableTargetNames,
+                new RangeShardingValue<>("t_order", "create_time", Range.closed("2020-01-01 00:00:11", "2020-01-01 00:00:21")));
+        assertThat(actualWithoutMilliseconds.size(), is(11));
```

Comment:
Could we add more test cases for `doSharding` to verify certain value's sharding target, but not only the target size? (I've added comment in PR about date parsing)

Reply:
> Could we add more test cases for `doSharding` to verify certain value's sharding target, but not only the target size? (I've added comment in PR about date parsing)

I added the corresponding instructions below the PR and removed this test.

Reply:
That's OK